### PR TITLE
Fix #318860: Restore rest offset with slash notation in voices 3 and 4

### PR DESCRIPTION
**Pull request by tobik**
Resolves: *https://musescore.org/en/node/318800#comment-1074094*

*Reintroduce offsets for rests in voice 3 and 4 (1-based) in connection with slash notation* 

According to https://musescore.org/en/node/318800#comment-1074063 slash notation in voice 1 and 2 are full size and not offset outside the stuff, voice 3 and 4 are small and outside the stuff.

- [x] I signed [CLA](https://musescore.org/en/cla)
- [x] I made sure the code in the PR follows [the coding rules](https://github.com/musescore/MuseScore/wiki/CodeGuidelines)
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"
- [x] I made sure the commit message title starts with "fix #424242:" if there is a related issue
- [ ] I created the test (mtest, vtest, script test) to verify the changes I made

### CODE REVIEW

**Review by Jojo-Schmitz — APPROVED** [review]

File: libmscore/rest.cpp [hunk 1]
```diff
@@ -532,7 +532,7 @@ int Rest::computeLineOffset(int lines)
                   }
             }
 
-      if (offsetVoices) {
+      if (offsetVoices && voice() < 2) {
```
Comment (Jojo-Schmitz) [inline libmscore/rest.cpp:535]: This means voice 1 and 2, right? IIRC `voice()` is 0 based.

Comment (tobik) [reply libmscore/rest.cpp:535]: Yes, don't offset voice 1 and voice 2 in combination with slash notation, but do offset voice 3 and 4

Comment (Jojo-Schmitz) [reply libmscore/rest.cpp:535]: Might be worth a comment explaining this?

Comment (tobik) [reply libmscore/rest.cpp:535]: For detailed description please see https://musescore.org/en/node/318800#comment-1074063

Comment (Jojo-Schmitz) [reply libmscore/rest.cpp:535]: Yes, that's where I found it. Might be good though to summarize this in a short comment in the code

**Comment by Jojo-Schmitz** [comment]
The 1st vtest fail shows an expected difference, and indeed needs fixing of the ref file.
The 2nd vtest fail is expected too, but for a different reason: it does not take the new/change ref file into account, but takes the one from the previous commit. So it is OK to have this vtest fail.

**Comment by tobik** [comment]
Then vtest will be OK with the next commit? One more reason to add the suggested remark :-)

**Comment by Jojo-Schmitz** [comment]
Not sure whether the vtests pick on the ref files of the previous commit, or the one the entire PR is based on.
But when in doubt, try it out ;-) and add a 3rd commit to add such a comment.
If it helps getting past the vtest fail, fine, if it doesn't, fine too, in any case I guess this PR should have only 1 commit in the end, and then would fail the vtest again. As we know about it, and verfied that the vtest fail is OK and has been dealt with properly, this won't be a problem as far as getting this PR merged. Except that it may need to get ported over to master (too).

**Review by Jojo-Schmitz — APPROVED** [review]

**Comment by Jojo-Schmitz** [comment]
OK, apparently the vtests use the last commit before the entire PR for comparison

**Review by MarcSabatella — APPROVED** [review]
Looks good to me!

**Comment by Jojo-Schmitz** [comment]
I guess you should squash these 3 commits into just one. I've ported it over to master meanwhile. see #8017

**Comment by vpereverzev** [comment]
merged into master, 3.x merges are closed

**Pull request closed without merging**